api/src/main/java/org/openmrs/api/impl/ProgramWorkflowServiceImpl.java

```diff
@@ -584,5 +586,43 @@ public ProgramWorkflow getWorkflow(Integer workflowId) {
 	public ProgramWorkflow getWorkflowByUuid(String uuid) {
 		return dao.getWorkflowByUuid(uuid);
 	}
-	
+        
+        @Override
+        public List<ProgramAttributeType> getAllProgramAttributeTypes() {
+            return dao.getAllProgramAttributeTypes();
+        }
+
+        @Override
+        public ProgramAttributeType getProgramAttributeType(Integer id) {
+            return dao.getProgramAttributeType(id);
+        }
+        
+        @Override
+        public ProgramAttributeType getProgramAttributeTypeByUuid(String uuid) {
+            return dao.getProgramAttributeTypeByUuid(uuid);
+        }
+
+        @Override
+        public ProgramAttributeType saveProgramAttributeType(ProgramAttributeType type) {
+            return dao.saveProgramAttributeType(type);
+        }
+
+        @Override
+        public void purgeProgramAttributeType(ProgramAttributeType type) {
+            dao.purgeProgramAttributeType(type);
+        }
```

Comment:
Where is getEncountersByPatientProgramUuid()?

Reply:
I removed it, I'm currently trying to write an implementation since their is none in the source